src/dune_engine/execution_parameters.mli

```diff
@@ -0,0 +1,32 @@
+(** Parameters that influence rule execution *)
+
+(** Such as:
+
+    - should target be set read-only?
+
+    - should aliases be expanded when sandboxing rules?
+
+    These often depend on the version of the Dune language used, which is
+    written in the [dune-project] file. Depending on the execution parameters
+    rather than the whole [dune-project] file means that when some part of the
+    [dune-project] file changes and this part does not have an effect on rule
+    execution, we can skip a bunch of work by not trying to re-execute all the
```

Comment:
As far as I can tell, `Execution_parameters` is not included into rule hash, anyway, so why would we be re-executing any rules when this changes?

Reply:
We are not going to re-execute the action, but we are still going to call `execute_rule`, which is going to re-evaluate the action builder, etc... In watch mode, I expect this to have a cost.

In the long run, it also feels like `Dune_project` shouldn't belong in `dune_engine`. `Execution_parameters` is a step forward in this direction.

Reply:
Oh, I get it. That makes sense.